### PR TITLE
Switch Fedora docker images to use quay.io

### PR DESCRIPTION
**Pull request by jay7x**
This should unbreak Fedora on aarch64.

### CODE REVIEW

File: lib/beaker-hostgenerator/hypervisor/docker.rb [hunk 1]
```diff
@@ -31,6 +31,8 @@ def image(ostype)
         image = ostype.sub(/(\d)/, ':\1')
 
         case ostype
+        when /^(fedora)/
```
Comment (ekohl) [inline lib/beaker-hostgenerator/hypervisor/docker.rb:34]:
```suggestion
        when /^fedora/
```

File: lib/beaker-hostgenerator/data.rb [hunk 2]
```diff
@@ -1160,10 +1160,7 @@ def generate_osinfo
       yield %w[aix73-POWER aix-7.3-power]
 
       # Fedora
-      (19..41).each do |release|
```
Comment (ekohl) [inline lib/beaker-hostgenerator/data.rb:1163]: As noted on IRC, there are official builds that still look maintained on Fedora 36.

**Review by ekohl — CHANGES_REQUESTED** [review]
I'm not sure this is the best way. I'm still not a fan of the `amd64/` prefix. 209336f52bf76fd103f86793c2b122d1884e580e explains it's specifically to run the x86_64 build on ARM chips like Apple.

This already gave issues for OpenSUSE (49526d17ee255760182669535ffa6ae3004e3721) so I wonder how we should deal with architectures.

If you want to use aarch64, I think you should use `fedora40-arm64` and not `fedora40-64`.

**Comment by jay7x** [comment]
Just realized I can override the `image` field in the setfile, so this change is not required for me right now. There is larger discussion might needed wrt the way we use the architecture.

**Pull request closed without merging**